include/boost/fusion/container/set/set.hpp

```diff
@@ -97,39 +97,20 @@ namespace boost { namespace fusion
 
         template <typename Sequence>
         BOOST_FUSION_GPU_ENABLED
-        set(Sequence const& rhs)
-            : data(rhs) {}
-
-#if !defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
-        template <typename Sequence>
-        BOOST_FUSION_GPU_ENABLED
         set(Sequence&& rhs)
             : data(std::forward<Sequence>(rhs)) {}
-#endif
 
         BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         explicit
         set(typename detail::call_param<T>::type ...args)
             : data(args...) {}
 
-#if !defined(BOOST_NO_CXX11_RVALUE_REFERENCES)
         template <typename ...U>
         BOOST_CXX14_CONSTEXPR BOOST_FUSION_GPU_ENABLED
         explicit
         set(U&& ...args)
```

Comment:
The situation I've been describing:
```c++
/* C++03 versions */
void foo(fusion::set<int> const&);
int main() {
    foo(10); // does not compile; implicit constructor is SFINAE disabled
}
```
This wasn't [always the case with the C++03 `fusion::set`](https://github.com/boostorg/fusion/commit/f872d132), but is likely the correct behavior. I think regression should be avoided here. The C++11 fusion list currently in master has the same implicit conversion issue.

> Yes. During constructing (or copying, moving, ...) nested container, outer container's constexpr ctor tries to instantiate inner container's ctor including implicit conversion ctor whcih has SFINAE, like enable_if<is_convertible<U0, T0>>::type, even if such ctor is not eligible.

I believe this was due to the implementation of `is_convertible`, and not specifically caused by the mixture of SFINAE and constexpr. If you remove `is_convertible` from the expression in the [prior bug report](https://svn.boost.org/trac/boost/ticket/11267), the code compiles. This does make SFINAE difficult, because the expression cannot invoke anything that might cause an unexpected evaluation/instantiation. `enable_if<is_sequence<Sequence>::value && sizeof...(T) == Sequence::size>` should work if `Sequence::size` is properly lazily evaluated, because the expression should never attempt to use any methods in `Sequence` or the target fusion object for evaluation.